runtime/misc.c

```diff
@@ -36,18 +36,26 @@ _Atomic caml_timing_hook caml_finalise_begin_hook = (caml_timing_hook)NULL;
 _Atomic caml_timing_hook caml_finalise_end_hook = (caml_timing_hook)NULL;
 
 #ifdef DEBUG
+void caml_print_loc(char_os *file_os, int line) {
+  char* file = caml_stat_strdup_of_os(file_os);
```

Comment:
This strikes me as a bad idea. If that `strdup` fails for some reason, it will raise an exception, which will either get handled and hide the problem we're attempting to report, or will not but whill hide it anyway...

Reply:
How would you suggest to do things differently? Note that this is exactly the same logic that would run previously on `CAMLassert(0)`.

Reply:
`char_os` is not necessarily the same as `char`, and this function performs the conversion from one to the other.

Reply:
Ah right, damned Windows support...